Description: Remove use of AS_VAR_GET
 This patch removes the use of AS_VAR_GET in the build system,
 which led to build failures.
Origin: upstream, https://svn.open-mpi.org/trac/ompi/changeset/23679
Author: Jeff Squyres <jsquyres@cisco.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=592892
Forwarded: https://svn.open-mpi.org/trac/ompi/ticket/2559
Last-Update: 2010-09-01

--- a/config/f90_get_sizeof.m4
+++ b/config/f90_get_sizeof.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -80,6 +81,6 @@
         unset happy ompi_conftest_h
         rm -rf conftest*])
 
-    $2=AS_VAR_GET(type_var)
+    AS_VAR_COPY([$2], [type_var])
     AS_VAR_POPDEF([type_var])dnl
 ])dnl
--- a/config/f77_get_sizeof.m4
+++ b/config/f77_get_sizeof.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -80,6 +81,6 @@
         unset happy ompi_conftest_h
         rm -rf conftest*])
 
-    $2=AS_VAR_GET(type_var)
+    AS_VAR_COPY([$2], [type_var])
     AS_VAR_POPDEF([type_var])dnl
 ])dnl
--- a/config/ompi_check_func_lib.m4
+++ b/config/ompi_check_func_lib.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -36,8 +37,8 @@
                  [AS_VAR_SET(ompi_var, "yes")],
                  [AS_VAR_SET(ompi_var, "not found")])
              LIBS="$LIBS_save"])])
-    AS_IF([test "AS_VAR_GET(ompi_var)" = "yes"],
-          [LIBS="$LIBS -l$2"])
+    AS_VAR_IF(ompi_var, [yes],
+              [LIBS="$LIBS -l$2"])
 
     # see if we actually have $1.  Use AC_CHECK_FUNCS so that it
     # does the glibc "not implemented" check.  Will use the current LIBS,
--- a/config/ompi_find_type.m4
+++ b/config/ompi_find_type.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -41,16 +42,12 @@
                [AS_VAR_SET(type_var, "not found")],
                [AS_VAR_SET(type_var, "$oft_real_type")])])
 
-    AS_IF([test "AS_VAR_GET(type_var)" = "not found"],
+    AS_VAR_IF(type_var, ["not found"],
           [AC_MSG_WARN([*** Did not find corresponding C type])
            AS_IF([test "$oft_abort_on_fail" != "no"],
                  [AC_MSG_ERROR([Cannot continue])])])
 
-    if test "AS_VAR_GET(type_var)" = "not found" ; then
-        $5=
-    else
-        $5=AS_VAR_GET(type_var)
-    fi
+    AS_VAR_IF(type_var, ["not found"], [$5=], [AS_VAR_COPY([$5], [type_var])])
 
     unset oft_real_type oft_target_size
 
--- a/config/f90_check_type.m4
+++ b/config/f90_check_type.m4
@@ -34,6 +34,6 @@
              [AS_VAR_SET(type_var, "no")])
          AC_LANG_POP([Fortran])])
 
-    AS_IF([test "AS_VAR_GET(type_var)" = "yes"], [$2], [$3])
+    AS_VAR_IF(type_var, [yes], [$2], [$3])
     AS_VAR_POPDEF([type_var])dnl
 ])dnl
--- a/config/f77_check_type.m4
+++ b/config/f77_check_type.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -34,7 +35,7 @@
              [AS_VAR_SET(type_var, "no")])
          AC_LANG_POP([Fortran 77])])
 
-    AS_IF([test "AS_VAR_GET(type_var)" = "yes"], [$2], [$3])
+    AS_VAR_IF(type_var, [yes], [$2], [$3])
     AS_VAR_POPDEF([type_var])dnl
 ])dnl
 
--- a/config/f90_get_int_kind.m4
+++ b/config/f90_get_int_kind.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -54,7 +55,7 @@
         unset happy ompi_conftest_h
         rm -rf conftest*])
 
-        $3=AS_VAR_GET(type_var)
+        AS_VAR_COPY([$3], [type_var])
     else
         $3=0
     fi
--- a/config/f77_check_real16_c_equiv.m4
+++ b/config/f77_check_real16_c_equiv.m4
@@ -93,7 +93,7 @@
         ])
     ])
 
-    ompi_real16_matches_c=AS_VAR_GET([real16_matches_c_var])
+    AS_VAR_COPY([ompi_real16_matches_c], [real16_matches_c_var])
     AS_VAR_POPDEF([real16_matches_c_var])
 
     AS_IF([test "$ompi_real16_matches_c" = "yes"],
--- a/config/f77_get_alignment.m4
+++ b/config/f77_get_alignment.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -90,7 +91,7 @@
                  AC_MSG_ERROR([Could not determine alignment of $1])])])
         rm -rf conftest*])
 
-    $2=AS_VAR_GET([type_var])
+    AS_VAR_COPY([$2], [type_var])
     AS_VAR_POPDEF([type_var])dnl
     OMPI_VAR_SCOPE_POP
 ])
--- a/config/ompi_lang_link_with_c.m4
+++ b/config/ompi_lang_link_with_c.m4
@@ -2,6 +2,7 @@
 dnl
 dnl Copyright (c) 2006      Los Alamos National Security, LLC.  All rights
 dnl                         reserved. 
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -62,6 +63,6 @@
      rm -f conftest_c.$ac_ext
      AC_LANG_POP(C)])
 
-  AS_IF([test "AS_VAR_GET([lang_var])" = "yes"], [$2], [$3])
+  AS_VAR_IF(lang_var, [yes], [$2], [$3])
   AS_VAR_POPDEF([lang_var])dnl
 ])
--- a/config/ompi_check_compiler_works.m4
+++ b/config/ompi_check_compiler_works.m4
@@ -5,6 +5,7 @@
 dnl                         Corporation.  All rights reserved.
 dnl Copyright (c) 2006      Los Alamos National Security, LLC.  All rights
 dnl                         reserved. 
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -37,7 +38,7 @@
                            [AS_VAR_SET(lang_var, ["links (cross compiling)"])],
                            [AS_VAR_SET(lang_var, ["no"])])])
          AC_LANG_POP($1)])
-    AS_IF([test "AS_VAR_GET(lang_var)" = "no"],
+    AS_VAR_IF(lang_var, [no], 
           [cat <<EOF >&2
 **********************************************************************
 * It appears that your $1 compiler is unable to produce working
--- a/config/f90_find_module_include_flag.m4
+++ b/config/f90_find_module_include_flag.m4
@@ -10,6 +10,7 @@
 dnl                         University of Stuttgart.  All rights reserved.
 dnl Copyright (c) 2004-2005 The Regents of the University of California.
 dnl                         All rights reserved.
+dnl Copyright (c) 2010      Cisco Systems, Inc.  All rights reserved.
 dnl $COPYRIGHT$
 dnl 
 dnl Additional copyrights may follow
@@ -71,7 +72,7 @@
             rm -rf conftest.$$
         ])
 
-        OMPI_FC_MODULE_FLAG=AS_VAR_GET(f90_inc_var)
+        AS_VAR_COPY([OMPI_FC_MODULE_FLAG], [f90_inc_var])
         if test "$OMPI_FC_MODULE_FLAG" = ""; then
         AC_MSG_WARN([*** Could not determine the f90 compiler flag to indicate where modules reside])
         AC_MSG_ERROR([*** Cannot continue])
